plugins/modules/hana_backup.py

```diff
@@ -0,0 +1,253 @@
+    comment: Created with ansible
+    binary_path: "/my/custom/path"
+    wait: false
+"""
+
+RETURN = r""" # """
+
+
+import os
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+def get_sql(module):
+    sql_wait = "" if module.params.get("wait") else "ASYNCHRONOUS"
+    backup_type = (
+        "" if module.params.get("type") == "FULL" else module.params.get("type")
+    )
+    sql_comment = (
+        "COMMENT '{0}'".format(module.params.get("comment"))
+        if module.params.get("comment")
+        else ""
+    )
+    sql_for_tenant = (
+        ""
+        if module.params.get("database_name") == "SYSTEMDB"
+        else "FOR {0}".format(module.params["database_name"])
+    )
+
+    sql_tooloption = ""
+    if module.params.get("backend") == "BACKINT":
+        sql_tooloption = (
+            "TOOLOPTION '{0}'".format(module.params.get("tooloption"))
+            if module.params.get("tooloption")
+            else ""
+        )
+
+    sql_prefix_destination = os.path.join(
+        module.params.get("destination"), module.params.get("prefix")
+    )
+    sql_command = "BACKUP DATA {0} {1} USING {2}('{3}') {4} {5} {6}".format(
+        backup_type,
+        sql_for_tenant,
+        module.params.get("backend"),
+        sql_prefix_destination,
+        sql_tooloption,
+        sql_wait,
+        sql_comment,
+    )
+
+    binary_path = os.path.join(module.params.get("binary_path"), "hdbsql")
+    args = [binary_path]
+
+    if module.params.get("host"):
+        args.extend(["-n", module.params.get("host")])
+
+    if module.params.get("instance_number"):
+        args.extend(["-i", module.params.get("instance_number")])
+
+    if module.params.get("hdbsqluserstore_key"):
+        args.extend(["-U", module.params.get("hdbsqluserstore_key")])
+    elif module.params.get("database_user_password"):
+        user = (
+            module.params.get("database_user")
+            if module.params.get("database_user")
+            else "SYSTEM"
+        )
+        args.extend(["-u", user, "-p", module.params.get("database_user_password")])
+
+    args.append(sql_command.strip())
+    return args
+
+
+def run_command(module, args):
+    return module.run_command(args=args)
+
+
+def ensure_backup_created(module):
+    if module.check_mode:
+        return True, "", ""
+
+    args = get_sql(module)
+    rc, stdout, err = run_command(module, args)
+    if rc != 0:
+        module.fail_json(cmd=args, rc=rc, stdout=stdout, msg=err)
+
+    return True, stdout, err, args
+
+
+def main():
+    module_args = dict(
+        host=dict(type="str"),
+        instance_number=dict(type="str"),
+        database_user_password=dict(type="str", no_log=True),
+        database_user=dict(type="str"),
+        hdbsqluserstore_key=dict(type="str", no_log=False),
+        database_name=dict(type="str", default="SYSTEMDB"),
+        type=dict(
+            type="str", choices=["INCREMENTAL", "DIFFERENTIAL", "FULL"], default="FULL"
+        ),
```

Comment:
should be outsourced to a own var outside the module_args to be clearer

Reply:
see: https://github.com/sap-linuxlab/community.sap_libs/blob/c4d27366c9b3b7f71f52409c25433f4f59c4843b/plugins/modules/sap_task_list_execute.py#L251 for example